### PR TITLE
chore: Refactor UnfoldResourceSource.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
Refactory UnfoldResourceSource a little instead of https://github.com/apache/pekko/pull/615

Result:
Clean code.

### CODE REVIEW

File: stream/src/main/scala/org/apache/pekko/stream/impl/UnfoldResourceSource.scala [hunk 1]
```diff
@@ -33,66 +34,69 @@ import pekko.stream.stage._
...
+        try {
+          readData(resource) match {
+            case Some(data) => push(out, data)
+            case None       => closeStage()
           }
+        } catch {
+          case NonFatal(ex) =>
+            decider(ex) match {
+              case Supervision.Stop =>
+                open = false
```
Comment (jxnu-liguobin) [inline stream/src/main/scala/org/apache/pekko/stream/impl/UnfoldResourceSource.scala:63]: indent

Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/UnfoldResourceSource.scala:63]: It's formatted with scala-cli

Comment (He-Pin) [reply stream/src/main/scala/org/apache/pekko/stream/impl/UnfoldResourceSource.scala:63]: You can review it with `whitespace:off` in the review tab.

**Review by Roiocam — APPROVED** [review]
LGTM

**Pull request merged**